lib/shared/widgets/givt_fab.dart

```diff
@@ -0,0 +1,132 @@
+import 'package:flutter/material.dart';
+
+class GivtFAButton extends StatefulWidget {
+  const GivtFAButton({
+    super.key,
+    this.isDisabled,
+    required this.onTap,
+    required this.text,
+    this.leftIcon,
+    this.rightIcon,
+  });
+
+  final VoidCallback onTap;
+  final bool? isDisabled;
+  final String text;
+
+  final Widget? leftIcon;
+  final Widget? rightIcon;
+  @override
+  _GivtFAButtonState createState() => _GivtFAButtonState();
+}
+
+class _GivtFAButtonState extends State<GivtFAButton> {
+  double dropShadowHeight = 4;
+  double paddingtop = 4;
+
+  @override
+  void initState() {
+    widget.isDisabled == true ? dropShadowHeight = 0 : dropShadowHeight = 4;
+    widget.isDisabled == true ? paddingtop = 4 : paddingtop = 0;
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: EdgeInsets.only(top: paddingtop),
+      child: GestureDetector(
+        onTap: widget.isDisabled == true ? null : widget.onTap,
+        onTapDown: widget.isDisabled == true
```

Comment:
Is it possible to reuse this logic? Looks like the same thing as the button above this one :)

Reply:
I tried to, but then all buttons have the same issue i mentioned in the slack thread, their state does not get initialized; 

https://github.com/givtnl/givt-app-kids/assets/43034761/2c3bf8e2-ee40-40db-94ea-072a3514892a

Reply:
https://github.com/givtnl/givt-app-kids/tree/feature/kids-130-share-button-logic 
This is where i was trying to share the logic